src/Array.mo

```diff
@@ -522,43 +525,47 @@ module {
     acc
   };
 
-  /// Flattens the array of arrays into a single array. Retains the original
+  /// Flattens the iterator of arrays into a single array. Retains the original
```

Comment:
I wonder if we should keep the original flatten, that can be achieved with a single allocation, and perhaps add an iterator  that can be implemented using that. 

The problem with this flatten is that you can't easily determine the size of the the array without iterating and storing the intermediate elements somewhere. Note you can't iterate twice to discover the total size and then again to append the elements,

Reply:
That's a good point. Maybe we could rename this to `flattenArray()` so that `flatten()` has a consistent API across different modules. I'll go ahead with this for now, and then we could revisit if it seems better for it to be the other way around.